### ISSUE TITLE
RPM package grafana-enterprise 8 obsoletes grafana 9

### ISSUE THREAD

**Issue by gytisrepecka** [grafana/grafana#62655]
**What happened**:

Doing `dnf update` on Red Hat Enterprise Linux suggests `grafana-enterprise` package intends to replace `grafana` (open source) package:

```plain
===============================================================================================================================================
 Package                                   Architecture                  Version                          Repository                      Size
===============================================================================================================================================
Installing:
 grafana-enterprise                        x86_64                        8.5.20-1                         grafana                         84 M
     replacing  grafana.x86_64 9.3.4-1

Transaction Summary
===============================================================================================================================================
Install  1 Package
```

This kind of issue was documented in 2020 May (#24801) and 2022 November (#58103) following [repository migration](https://grafana.com/docs/grafana/latest/setup-grafana/installation/rpm/) from `packages.grafana.com` to `rpm.grafana.com`.

**What you expected to happen**:

Package `grafana` (open source) is updated instead of offering `grafana-enterprise`.

**How to reproduce it (as minimally and precisely as possible)**:

Run `dnf update` having `grafana` (open source) package with version `9.3.4-1`.

Issue occurs with both `packages.grafana.com` and `rpm.grafana.com` repositories.

**Anything else we need to know?**:

**Environment**:
- Grafana version: v9.3.4 (ecf4e45659)
- OS Grafana is installed on: Red Hat Enterprise Linux release 8.6 (x86_64)

**Comment by robsliwi** [comment]
We also got quite some dependency hell now with marking `grafana` `obsolete`.

For example we're trying to upgrade `grafana` with `# dnf upgrade grafana` and getting the following output:

```
Error: 
 Problem: problem with installed package someotherpackage-1.0.5-1.el8.x86_64
  - package someotherpackage-1.0.5-1.el8.x86_64 requires grafana >= 8.4, but none of the providers can be installed
  - package someotherpackage-1.0.6-1.el8.x86_64 requires grafana >= 8.4, but none of the providers can be installed
  - package someotherpackage-1.0.7-1.el8.x86_64 requires grafana >= 8.4, but none of the providers can be installed
  - package someotherpackage-1.0.8-1.el8.x86_64 requires grafana >= 8.4, but none of the providers can be installed
  - package grafana-enterprise-8.5.20-1.x86_64 obsoletes grafana provided by grafana-9.3.2-1.x86_64
  - package grafana-enterprise-8.5.20-1.x86_64 obsoletes grafana provided by grafana-8.4.0-1.x86_64
  - package grafana-enterprise-8.5.20-1.x86_64 obsoletes grafana provided by grafana-8.4.1-1.x86_64
  - package grafana-enterprise-8.5.20-1.x86_64 obsoletes grafana provided by grafana-8.4.10-1.x86_64
  - package grafana-enterprise-8.5.20-1.x86_64 obsoletes grafana provided by grafana-8.4.11-1.x86_64
  - package grafana-enterprise-8.5.20-1.x86_64 obsoletes grafana provided by grafana-8.4.2-1.x86_64
  - package grafana-enterprise-8.5.20-1.x86_64 obsoletes grafana provided by grafana-8.4.3-1.x86_64
  - package grafana-enterprise-8.5.20-1.x86_64 obsoletes grafana provided by grafana-8.4.4-1.x86_64
  - package grafana-enterprise-8.5.20-1.x86_64 obsoletes grafana provided by grafana-8.4.5-1.x86_64
  - package grafana-enterprise-8.5.20-1.x86_64 obsoletes grafana provided by grafana-8.4.6-1.x86_64
  - package grafana-enterprise-8.5.20-1.x86_64 obsoletes grafana provided by grafana-8.4.7-1.x86_64
  - package grafana-enterprise-8.5.20-1.x86_64 obsoletes grafana provided by grafana-8.5.0-1.x86_64
  - package grafana-enterprise-8.5.20-1.x86_64 obsoletes grafana provided by grafana-8.5.1-1.x86_64
  - package grafana-enterprise-8.5.20-1.x86_64 obsoletes grafana provided by grafana-8.5.10-1.x86_64
  - package grafana-enterprise-8.5.20-1.x86_64 obsoletes grafana provided by grafana-8.5.11-1.x86_64
  - package grafana-enterprise-8.5.20-1.x86_64 obsoletes grafana provided by grafana-8.5.12-1.x86_64
  - package grafana-enterprise-8.5.20-1.x86_64 obsoletes grafana provided by grafana-8.5.13-1.x86_64
  - package grafana-enterprise-8.5.20-1.x86_64 obsoletes grafana provided by grafana-8.5.14-1.x86_64
  - package grafana-enterprise-8.5.20-1.x86_64 obsoletes grafana provided by grafana-8.5.15-1.x86_64
  - package grafana-enterprise-8.5.20-1.x86_64 obsoletes grafana provided by grafana-8.5.16-1.x86_64
  - package grafana-enterprise-8.5.20-1.x86_64 obsoletes grafana provided by grafana-8.5.2-1.x86_64
  - package grafana-enterprise-8.5.20-1.x86_64 obsoletes grafana provided by grafana-8.5.20-1.x86_64
  - package grafana-enterprise-8.5.20-1.x86_64 obsoletes grafana provided by grafana-8.5.3-1.x86_64
  - package grafana-enterprise-8.5.20-1.x86_64 obsoletes grafana provided by grafana-8.5.4-1.x86_64
  - package grafana-enterprise-8.5.20-1.x86_64 obsoletes grafana provided by grafana-8.5.5-1.x86_64
  - package grafana-enterprise-8.5.20-1.x86_64 obsoletes grafana provided by grafana-8.5.6-1.x86_64
  - package grafana-enterprise-8.5.20-1.x86_64 obsoletes grafana provided by grafana-8.5.9-1.x86_64
  - package grafana-enterprise-8.5.20-1.x86_64 obsoletes grafana provided by grafana-9.0.0-1.x86_64
  - package grafana-enterprise-8.5.20-1.x86_64 obsoletes grafana provided by grafana-9.0.1-1.x86_64
  - package grafana-enterprise-8.5.20-1.x86_64 obsoletes grafana provided by grafana-9.0.2-1.x86_64
  - package grafana-enterprise-8.5.20-1.x86_64 obsoletes grafana provided by grafana-9.0.3-1.x86_64
  - package grafana-enterprise-8.5.20-1.x86_64 obsoletes grafana provided by grafana-9.0.4-1.x86_64
  - package grafana-enterprise-8.5.20-1.x86_64 obsoletes grafana provided by grafana-9.0.5-1.x86_64
  - package grafana-enterprise-8.5.20-1.x86_64 obsoletes grafana provided by grafana-9.0.6-1.x86_64
  - package grafana-enterprise-8.5.20-1.x86_64 obsoletes grafana provided by grafana-9.0.7-1.x86_64
  - package grafana-enterprise-8.5.20-1.x86_64 obsoletes grafana provided by grafana-9.0.8-1.x86_64
  - package grafana-enterprise-8.5.20-1.x86_64 obsoletes grafana provided by grafana-9.0.9-1.x86_64
  - package grafana-enterprise-8.5.20-1.x86_64 obsoletes grafana provided by grafana-9.1.0-1.x86_64
  - package grafana-enterprise-8.5.20-1.x86_64 obsoletes grafana provided by grafana-9.1.1-1.x86_64
  - package grafana-enterprise-8.5.20-1.x86_64 obsoletes grafana provided by grafana-9.1.2-1.x86_64
  - package grafana-enterprise-8.5.20-1.x86_64 obsoletes grafana provided by grafana-9.1.3-1.x86_64
  - package grafana-enterprise-8.5.20-1.x86_64 obsoletes grafana provided by grafana-9.1.4-1.x86_64
  - package grafana-enterprise-8.5.20-1.x86_64 obsoletes grafana provided by grafana-9.1.5-1.x86_64
  - package grafana-enterprise-8.5.20-1.x86_64 obsoletes grafana provided by grafana-9.1.6-1.x86_64
  - package grafana-enterprise-8.5.20-1.x86_64 obsoletes grafana provided by grafana-9.1.7-1.x86_64
  - package grafana-enterprise-8.5.20-1.x86_64 obsoletes grafana provided by grafana-9.1.8-1.x86_64
  - package grafana-enterprise-8.5.20-1.x86_64 obsoletes grafana provided by grafana-9.2.0-1.x86_64
  - package grafana-enterprise-8.5.20-1.x86_64 obsoletes grafana provided by grafana-9.2.1-1.x86_64
  - package grafana-enterprise-8.5.20-1.x86_64 obsoletes grafana provided by grafana-9.2.10-1.x86_64
  - package grafana-enterprise-8.5.20-1.x86_64 obsoletes grafana provided by grafana-9.2.2-1.x86_64
  - package grafana-enterprise-8.5.20-1.x86_64 obsoletes grafana provided by grafana-9.2.3-1.x86_64
  - package grafana-enterprise-8.5.20-1.x86_64 obsoletes grafana provided by grafana-9.2.4-1.x86_64
  - package grafana-enterprise-8.5.20-1.x86_64 obsoletes grafana provided by grafana-9.2.5-1.x86_64
  - package grafana-enterprise-8.5.20-1.x86_64 obsoletes grafana provided by grafana-9.2.6-1.x86_64
  - package grafana-enterprise-8.5.20-1.x86_64 obsoletes grafana provided by grafana-9.2.7-1.x86_64
  - package grafana-enterprise-8.5.20-1.x86_64 obsoletes grafana provided by grafana-9.2.8-1.x86_64
  - package grafana-enterprise-8.5.20-1.x86_64 obsoletes grafana provided by grafana-9.3.0-1.x86_64
  - package grafana-enterprise-8.5.20-1.x86_64 obsoletes grafana provided by grafana-9.3.1-1.x86_64
  - package grafana-enterprise-8.5.20-1.x86_64 obsoletes grafana provided by grafana-9.3.4-1.x86_64
  - package grafana-enterprise-8.5.20-1.x86_64 obsoletes grafana provided by grafana-9.3.6-1.x86_64
  - cannot install the best update candidate for package grafana-9.3.2-1.x86_64
(try to add '--skip-broken' to skip uninstallable packages or '--nobest' to use not only best candidate packages)
```

**Comment by Daryes** [comment]
Same problem here on RHEL8 vith grafana oss v9.1.8, results are the same with either packages.grafana.com/oss/rpm or rpm.grafana.com  
I tried to completely  remove the repo and clean up the keys, as per the issue #62544 that might be related, but alas, no change, only the enterprise version is proposed.

The most surprising thing is the switch to the enterprise version while the repo was using oss as a target in the folder path.

**Comment by laugmanuel** [comment]
Same for us.

**Comment by gytisrepecka** [comment]
> The most surprising thing is the switch to the enterprise version while the repo was using oss as a target in the folder path.

Indeed - I've also had `packages.grafana.com/oss/rpm` - full path to open source version, but regardless `grafana-enterprise` was offered.

**Comment by hunter-nl** [comment]
Add ```exclude=grafana-enterprise*``` to /etc/yum.repo.d/grafana.repo file to ignore the enterprise version.

**Comment by gytisrepecka** [comment]
> Add `exclude=grafana-enterprise*` to /etc/yum.repo.d/grafana.repo file to ignore the enterprise version.

Thanks, this works - successfully upgraded to open source version `9.3.6`.

**Comment by digdilem** [comment]
Thanks for the workaround @hunter-nl, this issue also killed updates on one of my machines but with your help we were able to continue.

**Comment by markcichy** [comment]
Thank you @hunter-nl !!! Saved my many servers from 'dependency hell' mentioned above. I had forced the enterprise upgrade thinking, "how bad could it be," not realizing that "upgrading" to enterprise would kill my `pcp-rdis` plugin. 🤦🏻‍♂️
Thankfully you posted a couple of days ago, so I was able to roll back, reinstall the plugin, and exclude `grafana-enterprise` from `dnf update` - simple fix, VERY helpful.
